shared/src/vyxal/MiscHelpers.scala

```diff
@@ -199,4 +203,18 @@ object MiscHelpers:
       case (a: VList, b: VList) => VList.from((a, b).zipped.map(subtract(_, _)))
       case (a, b) => ???
 
+  def untilNoChange(function: VFun, value: VAny)(using Context): VList =
+    var prev = value
+    val res = LazyList.unfold(value) { curr =>
+      val next = function(curr)
+      if next == prev then None
+      else
+        prev = next
+        Some(next -> next)
+    }
+    VList.from(value #:: res)
+
+  def zipWith(left: VList, right: VList, function: VFun)(using Context): VList =
```

Comment:
Since there's already a method for zipping `VList`s that works on Scala functions, I think we can add methods to `VFun` for converting to Scala functions (`.asFn1`, `.asFn2`, ...). That'd avoid duplicating code for Scala functions and `VFun`s. Thoughts? @Lyxal

Reply:
That might be a bit tricky as `VFun` would need all possible `asFn<arity>` functions. Also, how would `asFnNeg1` work?

Reply:
We wouldn't make an `asFnNeg1`. It's only so that `VFun`s can be used with methods that want a `VAny => Context ?=> VAny` or `(VAny, VAny) => Context ?=> VAny`. `asFn1` and `asFn2` should be adequate, maybe `asFn3` at most.